presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -238,6 +239,28 @@ public static long hashCode(@SqlType(StandardTypes.DOUBLE) double value)
     }
 
     @ScalarOperator(SATURATED_FLOOR_CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long saturatedFloorCastToFloat(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        float result;
+        float minFloat = -1.0f * Float.MAX_VALUE;
+        if (value <= minFloat) {
+            result = minFloat;
+        }
+        else if (value >= Float.MAX_VALUE) {
+            result = Float.MAX_VALUE;
+        }
+        else {
+            result = (float) value;
```

Comment:
Yes, that should be correct. strictfp may have a non-trivial performance hit. It probably doesn't matter for how this function is used, but I'd be curious about the impact.